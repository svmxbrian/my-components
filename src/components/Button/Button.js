import React from 'react'
import PropTypes from 'prop-types'

const Button = ({ text, onClick }) => (
  <input class="button" type="button" onClick={onClick} value={text} />
)

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
}

export default Button
