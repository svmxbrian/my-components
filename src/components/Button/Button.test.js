import React from 'react'
import Button from './Button'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('Button', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Button text="" onClick={() => {}} />)
    expect(wrapper).toBeDefined()
  })

  it('renders correctly', () => {
    const tree = renderer.create(<Button text="" onClick={() => {}} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  const mockFn = jest.fn()
  const text = 'text here'
  const wrapper = shallow(<Button text={text} onClick={mockFn} />)

  it('should call onClick function when it is clicked', () => {
    wrapper.find('input').simulate('click')
    expect(mockFn.mock.calls.length).toBe(1)
  })

  it('should show text on button', () => {
    // Get value by using wrapper props value
    expect(wrapper.find('input').props().value).toEqual(text)
  })
})
