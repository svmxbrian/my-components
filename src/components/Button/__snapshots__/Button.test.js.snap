// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<input
  class="button"
  onClick={[Function]}
  type="button"
  value=""
/>
`;
